lib/util/heap-profiler.js

```diff
@@ -0,0 +1,31 @@
+const inspector = require('inspector');
+const fs = require('fs').promises;
+
+class HeapProfiler {
+    constructor () {
+        this.session = new inspector.Session();
+        this.file = null;
+    }
+
+    async connect() {
+        this.session.connect();
+
+        this.file = await fs.open(`HeapSnapshot-${process.pid}-${Date.now()}.heapsnapshot`, 'w');
```

Comment:
where will this file be created? ./? or ./?lib/util/heapprofiler? or maybe wherever connect() is called?

Reply:
on current working directory of Node.js process

Reply:
in our case, main application directory ;)